File: mirea/cources/2026-2027/aii-july/03/contest/B.cpp
#include <bits/stdc++.h>
using namespace std;
void solve() {
    // читаем элементы и запоминаем их позиции в map:
    int n, q; cin >> n >> q;
    map<int, int> firstPos, lastPos;
    for (int i = 1; i <= n; i++) {
        int x; cin >> x;
        // ищем последнюю позицию
        auto iter = lastPos.find(x);
        if (iter == lastPos.end()) {
            // не найдена - элемент ещё не встречался
            firstPos[x] = i;
            lastPos[x] = i;
        } else {
            // найдена - меняем только последнюю позицию
            iter->second = i;
        }
    }
    // читаем и отвечаем на запросы:
    while (q --> 0) {
        int x; cin >> x;
        auto iter = lastPos.find(x);
        if (iter == lastPos.end())
            cout << "-1\n";
        else // найден:
            cout << firstPos[x] << ' ' << iter->second << '\n';
    }
}
main() {
    int tt; cin >> tt;
    while (tt --> 0) solve();
}

File: mirea/cources/2026-2027/aii-july/03/contest/B.py
def solve():
    # читаем элементы и запоминаем их позиции в map:
    n, q = map(int, input().split())
    firstPos, lastPos = {}, {}
    i = 1
    while i <= n:
        for x in map(int, input().split()):
            # ищем последнюю позицию
            if x not in lastPos:
                # не найдена - элемент ещё не встречался
                firstPos[x] = i
                lastPos[x] = i
            else:
                # найдена - меняем только последнюю позицию
                lastPos[x] = i
            i += 1
            if i > n:
                break
    # читаем и отвечаем на запросы:
    while q > 0:
        for x in map(int, input().split()):
            if x not in lastPos:
                print(-1)
            else:  # найден:
                print(firstPos[x], lastPos[x])
            q -= 1
            if q == 0:
                break


tt = int(input())
for _ in range(tt):
    solve()